From 1ef1bb2f1b106ce29f43d7df609b9a0de5a54f9d Mon Sep 17 00:00:00 2001
From: Lawrence Rust <lvr@softsystem.co.uk>
Date: Tue, 8 Nov 2011 19:55:28 +0100
Subject: [PATCH 073/285] mythuibuttonlist: Prevent reading beyond end of list

Prevent a potential SEGV in MythUIButtonList::SetPosition by testing
if an iterator is < end() before dereferencing.

This was picked by by valgrind.

Signed-off-by: Lawrence Rust <lvr@softsystem.co.uk>
---
 mythtv/libs/libmythui/mythuibuttonlist.cpp |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/mythtv/libs/libmythui/mythuibuttonlist.cpp b/mythtv/libs/libmythui/mythuibuttonlist.cpp
index 0690b50..a5d7fd4 100644
--- a/mythtv/libs/libmythui/mythuibuttonlist.cpp
+++ b/mythtv/libs/libmythui/mythuibuttonlist.cpp
@@ -1239,7 +1239,7 @@ void MythUIButtonList::SetPosition(void)
     if (it < m_itemList.begin())
         it = m_itemList.begin();
 
-    int curItem = GetItemPos(*it);
+    int curItem = it < m_itemList.end() ? GetItemPos(*it) : 0;
     while (it < m_itemList.end() && button < (int)m_itemsVisible)
     {
         realButton = m_ButtonList[button];
-- 
1.7.9.5

